### ISSUE TITLE
Unable to set variations of fonts despite being able to select them

### ISSUE THREAD

**Issue by alemvigh** [vkbo/novelWriter#1733]
I'm not sure if this is a bug or a feature, but I'm unable to set a different font weight as the default font for the editor.

My use case is that Regular is usually too light, and Bold is usually too heavy, so I tend to go for the Medium variants.

As an example, under **Document Style > Document font family**, I can click the **...** button which will list different variations of the installed fonts such as "**Regular**", "**Light**" and "**Medium**", but clicking any of these followed by **Select** will just revert to the "**Regular**" version of the selected font.

I suspect this might be by design due to the wording **Document font family** instead of **Document font**, but in that case only listing the font family and not the different variations might make more sense.

Also the window is called **Select font** which is inconsistent with **Select font family** in the parent window.

![image](https://github.com/vkbo/novelWriter/assets/162419588/3830bb0f-3a91-4ba0-8dba-0756a5f956bf)

![image](https://github.com/vkbo/novelWriter/assets/162419588/d27032ff-783e-4c80-9013-8d1a10a4edad)

I think the solution could be either of these, even though I'd prefer the first one due to my use case:

**Selecting font variations **_should_** be possible:**
* Enable the selection of variations of fonts
* Change the wording in **Document Style > Document font family** to **Document font**

**Selecting font variations should **_not_** be possible:**
* Only display font family names in the selection box
* Change the name of the selection box from **Select Font** to **Select font family** for consistency

**Comment by alemvigh** [comment]
For the records I worked around this by selecting a font family where I intentionally only had installed the Medium weight so it had no choice but to use it when I selected the family.

**Comment by zxygentoo** [comment]
Met with the same confusion and found this one. I also like the first solution better: these days people have more and more fonts with good variation/weight support on their systems, and they might like to use them (like me :joy: )

**Comment by vkbo** [comment]
It's the OS font selection tool that provides multiple style options, not the app itself. I'm not sure how to transfer that information to the app yet.

**Comment by vkbo** [comment]
There are a ton of font settings returned by the font selector dialogs. I found out that Qt has a way to serialise all of those into a string, which it can take in return as a single setting value. I've implemented this in #1881. Unfortunately, perhaps, that also includes the font size, so I've removed the additional font size settings. That means everyone will have their font size reset to whatever the default is the first time they launch after 2.5. Hopefully that isn't too much of a disruption. It will, however, keep the font family setting since I'm reusing that same field.

**Comment by vkbo** [comment]
I noticed that the font stretch setting isn't being preserved. That is, "condensed", "expanded", etc. I can load a font with any of these settings, but the stretch factor is always 0 when I extract it to save it for the next session, so it returns to the default value. The built-in serialisation of the settings ignores the value in Qt5, so I can only assume there is an issue with it in the code. The value *is* serialised in Qt6, so maybe it has been fixed there.
